### PR TITLE
EVAKA-4455 Allow marking new staff attendances in the staff attendance table on group views

### PR DESCRIPTION
**Pull request by zch**
Also includes a commit for optimizing fetching of groups for employees.

Might be easier to review by commit than all at once.

### CODE REVIEW

File: frontend/src/employee-frontend/components/unit/unit-reservations/StaffAttendanceTable.tsx [hunk 1]
```diff
@@ -306,8 +315,41 @@ const AttendanceRow = React.memo(function AttendanceRow({
       <StyledTd partialRow={false} rowIndex={rowIndex} rowSpan={1}>
         {editing ? (
           <EditStateIndicator
-            status={saveRequestStatus}
-            stopEditing={() => setEditing(false)}
+            status={Success.of()}
+            stopEditing={() => {
+              setEditing(false)
+              const savePromises = values.flatMap(({ date, timeRanges }) =>
```
Comment (zch) [inline frontend/src/employee-frontend/components/unit/unit-reservations/StaffAttendanceTable.tsx:321]: It might make sense to refactor the endpoint to accept a list of attendances to upsert, but I'd rather do that in a separate PR I think.

File: service/src/main/kotlin/fi/espoo/evaka/attendance/StaffAttendanceQueries.kt [hunk 2]
```diff
@@ -277,13 +278,19 @@ fun Database.Read.getExternalStaffAttendancesByDateRange(unitId: DaycareId, rang
...
-fun Database.Read.getGroupsForEmployee(employeeId: EmployeeId): List<GroupId> = createQuery(
+private data class EmployeeGroups(
+    val employeeId: EmployeeId,
+    @Json
+    val groupIds: List<GroupId>
+)
+fun Database.Read.getGroupsForEmployees(employeeIds: Set<EmployeeId>): Map<EmployeeId, List<GroupId>> = createQuery(
     """
-    SELECT daycare_group_id
+    SELECT employee_id, json_agg(daycare_group_id) AS group_ids
```
Comment (akheron) [inline service/src/main/kotlin/fi/espoo/evaka/attendance/StaffAttendanceQueries.kt:288]: Contrary to what I previously said `array_agg(...)` would probably be more readable here. Then you can also remove the `@Json` annotation above.

File: frontend/src/employee-frontend/components/unit/unit-reservations/StaffAttendanceTable.tsx [hunk 3]
```diff
@@ -328,8 +315,41 @@ const AttendanceRow = React.memo(function AttendanceRow({
...
```
Comment (akheron) [inline frontend/src/employee-frontend/components/unit/unit-reservations/StaffAttendanceTable.tsx:321]: This could be just `values.map(...)` because you're returning a promise from the function in every case. Or you could replace `return Promise.resolve()` in the end with `return []` to take full advantage of the flatmap.

**Review by akheron — APPROVED** [review]

File: frontend/src/employee-frontend/components/unit/unit-reservations/StaffAttendanceTable.tsx (hunk 3 above)
Comment (zch) [reply frontend/src/employee-frontend/components/unit/unit-reservations/StaffAttendanceTable.tsx:321]: Actually the map on the following row returns an array of promises, which most often contains only one promise, but may contain several in case the employee has several attendance markings for the same date.
